### DISCUSSION TITLE
Combat screenshot

### DISCUSSION THREAD

**Discussion by pruuk** [evennia/evennia#2411] [My Evennia Project]
I still have a lot of work to do, but I have a twitch-style combat system up and working in my game. I thought I'd share some screenshots. On the left is the web client window with a third-party observer's view of the combat (my superuser is in the room). On the right is the combat from the first person POV in a TinTin+ window:
![Screenshot (3)](https://user-images.githubusercontent.com/72707866/117393065-8b54ce00-aeb0-11eb-8439-79dc82bc7263.png)

orange text is a hit text string
cyan is a dodge
green is a block
highlighted white text is the manuevering/footwork phase before a hit attempt.

**Comment by Griatch** [comment]
That looks plenty nice, @pruuk (the combat log works fine too it seems!). 
Is Tintin++ hiding your input though? Or is all the stuff after `attack guard` generated without further input? Stuff like `Meirok has no interest in grappling...` seems very specific and like it would the result of some some sort of command from you ...?

One thing to keep in mind when relaying info is that people with limited/no eyesight often play MUDs. They can read text very quickly (much quicker than you may expect). But they don't see your colors. So consider (maybe as an option) prefixing lines carrying particular info, with tags like `[hit]`, `[dodge]`, `[block]` and `[footwork]` so as to not only carry that info in color.

**Reply by pruuk** [reply]
Prefacing the lines with tags is a great idea. I'll put that in. :)

All of the stuff is generated without further input, though you can choose to be more active and change your fighting style as the fight goes along. One of the default attacks is 'grapple'. In this case, Meirok had rolled a super high Dexterity roll and the Guard has a very low Dex (and very low unarmed combat skill to boot), so he is pretty bad at grappling and it is set as his default attack. In the lines above, he tries a takedown and fails so badly that he ends up on the bottom. Meirok's default attack is with a weapon, so that line 'Meirok has no interest in grappling...' is the message that he is going to let Guard back to his feet w/o resistance.

The different attacks you can choose as a default currently are: 'unarmed', 'grapple', 'weapon', 'bash', 'defend', 'feint', 'yield', 'flee', 'mercy' and 'taunt'. 'ranged' will be added next. It's actually a pretty viable strategy in this combat setup to defend or grapple until your opponent just runs out of Stamina (which also ends combat). There are a toggle-able set of attributes called 'Mercy', 'Yield', and 'Wimpy' that allow you to show mercy to an opponent that is low on health/stamina, attempt to auto-yield, or attempt to auto-flee when your health/stamina is low(wimpy).

Under the hood, the combat is calling a function on the character/npc to get combat modifiers --> footwork or groundwork check at the start of each combatant's "turn" --> determining how many attacks --> execute an attack/yield/etc --> call a messaging function --> msg func calls a text dictionary file with first and third person strings --> msg func messages each set of folks in the room differently depending if they are the attacker, defender, or an observer --> next attack action or next combatant's turn.

My plan once I get this cleaned up and the building tools done is to make a newbie school area to explain the mechanics of combat in more detail without breaking the fourth wall too much.

**Reply by Griatch** [reply]
I see. Thanks for the explanation! I tend to prefer a more turn-based approach myself, but the output you get here is quite compelling. :+1: 
.
Griatch
